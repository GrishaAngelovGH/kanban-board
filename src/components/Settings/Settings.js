import { useState } from "react"

import ListGroup from "react-bootstrap/ListGroup"

import Modal from "components/Modal"

import settingsRepository from "persistent/persistentSettingsRepository"

const Settings = ({ show, onClose, onConfirm }) => {
  const [background, setBackground] = useState(settingsRepository.getBackground())

  const handleBackgroundClick = ({ target }) => {
    setBackground(target.innerText)
  }

  const handleConfim = () => {
    onConfirm({
      background
    })
  }

  const backgrounds = ["Nature Background", "Geometric Background", "No Background"]

  return (
    <Modal
      show={show}
      onClose={onClose}
      onConfirm={handleConfim}
      title="Settings"
      body={
        <ListGroup>
          <ListGroup.Item className="fw-bold bg-secondary-subtle">
            Background
          </ListGroup.Item>
          {
            backgrounds.map((v, i) => (
              <ListGroup.Item
                key={i}
                action
                onClick={handleBackgroundClick}
                className="d-flex justify-content-between align-items-center"
                style={{ height: 50 }}
              >
                <span>{v}</span>
                {v === background && <i className="bi bi-check-circle-fill text-primary fs-5"></i>}
              </ListGroup.Item>
            ))
          }
        </ListGroup>
      }
    />
  )
}

export default Settings