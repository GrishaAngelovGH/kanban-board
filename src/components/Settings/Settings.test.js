import { render } from "@testing-library/react"

import Settings from "./Settings"

test("should render Settings component", () => {
  const view = render(<Settings show />)

  expect(view).toMatchSnapshot()
})