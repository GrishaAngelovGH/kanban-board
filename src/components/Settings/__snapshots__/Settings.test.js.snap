// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Settings component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    class="modal-open"
    data-rr-ui-modal-open=""
    style="padding-right: 1024px;"
  >
    <div />
    <div
      class="fade modal-backdrop show"
    />
    <div
      aria-modal="true"
      class="fade modal show"
      role="dialog"
      style="display: block; padding-right: 0px;"
      tabindex="-1"
    >
      <div
        class="modal-dialog"
      >
        <div
          class="modal-content"
        >
          <div
            class="modal-header"
          >
            <div
              class="modal-title h4"
            >
              Settings
            </div>
          </div>
          <div
            class="modal-body"
          >
            <div
              class="list-group"
            >
              <div
                class="fw-bold bg-secondary-subtle list-group-item"
              >
                Background
              </div>
              <button
                class="d-flex justify-content-between align-items-center list-group-item list-group-item-action"
                style="height: 50px;"
              >
                <span>
                  Nature Background
                </span>
              </button>
              <button
                class="d-flex justify-content-between align-items-center list-group-item list-group-item-action"
                style="height: 50px;"
              >
                <span>
                  Geometric Background
                </span>
                <i
                  class="bi bi-check-circle-fill text-primary fs-5"
                />
              </button>
              <button
                class="d-flex justify-content-between align-items-center list-group-item list-group-item-action"
                style="height: 50px;"
              >
                <span>
                  No Background
                </span>
              </button>
            </div>
          </div>
          <div
            class="modal-footer"
          >
            <button
              class="btn btn-secondary"
              type="button"
            >
              Close
            </button>
            <button
              class="btn btn-primary"
              type="button"
            >
              Confirm
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
